### ISSUE TITLE
Jump to source code from 3D preview - make selected line more visible

### ISSUE THREAD

**Issue by gingerbeardman** [openscad/openscad#3664]
## Steps
1. Right click a model
2. Choose a shape to jump to source
3. Look at source window for selected line

## Expected
- line should be centred in source window for maximum accessibility/visibility

## Actual
- selected line can appear anywhere in the source window
- most commonly it is at top/bottom depending on whether the line was before/after where you were in the source before using the feature

## Screenshots

- target line was later in source file than current line before calling jump to source
  <img width="1080" alt="Screen shot 2021-02-12 at 13 26 32" src="https://user-images.githubusercontent.com/49612/107773692-eeade480-6d35-11eb-9856-9fa279f5b3df.png">

- target line was earlier in source file than current line before calling jump to source
  <img width="1080" alt="Screen shot 2021-02-12 at 13 26 24" src="https://user-images.githubusercontent.com/49612/107773680-e9509a00-6d35-11eb-8f89-ab6e0b4faae4.png">

- target line on same screen of source as current line before calling jump to source
  <img width="1080" alt="Screen shot 2021-02-12 at 13 28 17" src="https://user-images.githubusercontent.com/49612/107773867-2cab0880-6d36-11eb-8e2f-c0b975b92d9c.png">

**Comment by Tomatower** [comment]
First of all - yay someone is using the feature :) 

Second, this "Move Cursor" codepath is shared with several other Cursor movement thingies. So I would fix that by making sure to have +- 3 lines in view.

So when you have 100 lines in view, and your search jumps *down* to line 200, you would see the window 103-203.
Jumping *up* would bring lines 197-297 into view. 
This is how VIM works in its default config, and IMO the least intrusive.

**Comment by t-paul** [comment]
Yes, I agree it's probably better to have that kind of threshold so the code does not jump around if the target is already visible near the center. We'll have to check what qscintilla supports.

I would like to see an additional highlight too if we can find a way, depending on the color scheme it might be not too obvious where the cursor ended up. I don't know what's possible, some kind of graphical effect like flashing the line or the line number.

**Comment by Tomatower** [comment]
QScintilla has 

> `void QsciScintilla::ensureLineVisible(int line)`
> Ensures that the line number line is visible. 


[Source](http://gnu-darwin.org/www001/ports-1.5a-CURRENT/devel/qscintilla2/work/QScintilla-1.73-gpl-2.1/doc/html-Qt4/classQsciScintilla.html#1d0ab66e8831b88f6a876642c896b6ef)

So some boundaries check and testing at the edge cases (going past the end/beginning) sound feasible.

**Comment by gingerbeardman** [comment]
Thanks all!

This is a great update.

**Comment by Tomatower** [comment]
See linked PR, this works except for the first and last line - I personally do not like overscrolling.
